src/main/java/org/jboss/tm/TransactionLocal.java

```diff
@@ -26,296 +26,257 @@
-    * Typical implementations of initialValue will call an appropriate
-    * constructor and return the newly constructed object.
-    *
-    * @return the initial value for this TransactionLocal
-    */
-   protected Object initialValue()
-   {
-      return null;
-   }
-
-
-   /**
-    * get the transaction local value.
-    *
-    * @param tx the transaction
-    * @return the obejct
-    */
-   protected Object getValue(Transaction tx)
-   {
-      return delegate.getValue(this, tx);
-   }
-
-   /**
-    * put the value in the TransactionImpl map
-    *
-    * @param tx the transaction
-    * @param value the value
-    */
-   protected void storeValue(Transaction tx, Object value)
-   {
-      delegate.storeValue(this, tx, value);
-   }
-
-   /**
-    * does Transaction contain object?
-    *
-    * @param tx the transaction
-    * @return true if it has an object
-    */
-   protected boolean containsValue(Transaction tx)
-   {
-      return delegate.containsValue(this, tx);
-   }
-
-   /**
-    * Returns the value of this TransactionLocal variable associated with the
-    * thread context transaction. Creates and initializes the copy if this is
-    * the first time the method is called in a transaction.
-    *
-    * @return the value of this TransactionLocal
-    */
-   public Object get()
-   {
-      return get(getTransaction());
-   }
-
-
-   /**
-    * Returns the value of this TransactionLocal variable associated with the
-    * specified transaction. Creates and initializes the copy if this is the
-    * first time the method is called in a transaction.
-    *
-    * @param transaction the transaction for which the variable it to
-    * be retrieved
-    * @return the value of this TransactionLocal
-    * @throws IllegalStateException if an error occures while registering
-    * a synchronization callback with the transaction
-    */
-   public Object get(Transaction transaction)
-   {
-      if (transaction == null) return initialValue();
-
-      Object value = getValue(transaction);
-
-      // is we didn't get a value initalize this object with initialValue()
-      if(value == null)
-      {
-         // get the initial value
-         value = initialValue();
-
-         // if value is null replace it with the null value standin
-         if(value == null)
-         {
+public class TransactionLocal {
+
+    /**
+     * To simplify null values handling in the preloaded data pool we use this value instead of
+     * 'null'
+     */
+    private static final Object NULL_VALUE = new Object();
+
+    /**
+     * The transaction manager is maintained by the system and manges the assocation of transaction
+     * to threads.
+     */
+    protected final TransactionManager transactionManager;
+
+    /** The delegate */
+    protected TransactionLocalDelegate delegate;
+
+    /**
+     * Creates a thread local variable.
+     *
+     * @throws IllegalStateException if there is no system transaction manager
```

Comment:
Since an IllegalStateException is unchecked, it does not need to be declared in the throws clause of the constructor. I'd expect the same to apply to the Javadoc too, perhaps it could be drawn out in the description but *not* defined formally in the javadoc. So at some point we need to correct things like this.